jetty-websocket/websocket-javax-client/src/main/java/org/eclipse/jetty/websocket/javax/client/internal/JavaxWebSocketClientContainer.java

```diff
@@ -271,4 +278,68 @@ private ClientEndpointConfig getAnnotatedConfig(Object endpoint) throws Deployme
 
         return new AnnotatedClientEndpointConfig(anno);
     }
+
+    @Override
+    protected void doStart() throws Exception
+    {
+        doClientStart();
+        super.doStart();
+    }
+
+    protected void doClientStart()
+    {
+        try
+        {
+            Object context = getClass().getClassLoader()
+                .loadClass("org.eclipse.jetty.server.handler.ContextHandler")
+                .getMethod("getCurrentContext")
+                .invoke(null);
+
+            Object contextHandler = context.getClass()
+                .getMethod("getContextHandler")
+                .invoke(context);
+
+            contextHandler.getClass()
+                .getMethod("addManaged", LifeCycle.class)
```

Comment:
As @sbordet says above, I don't think we can add `this` as a `LifeCycle` instance, because it may be a different `LifeCycle`.

Two options:
1. create a Proxy class from the LifeCycle interface obtained from the same classloader as the ContextHandler
2. use a javax.servlet.ServletContextListener... but @janbartel will have to let us add one after init and for it's destroy method to still be called (see #5837 )